ibridges/cli/config.py

```diff
@@ -0,0 +1,218 @@
+            if str(DEFAULT_IENV_PATH) not in self.servers:
+                raise ValueError("Default iRODS path not in configuration file.")
+            if not isinstance(self.cur_env, str):
+                raise ValueError(f"Current environment should be a string not {type(self.cur_env)}")
+            cur_aliases = set()
+            new_servers = {}
+            for ienv_path, entry in self.servers.items():
+                if ienv_path != str(DEFAULT_IENV_PATH) and not Path(ienv_path).is_file():
+                    warnings.warn(f"Environment with file '{ienv_path}' does not exist anymore, "
+                                   "removing the entry.")
+                elif entry.get("alias", None) in cur_aliases:
+                    warnings.warn(f"Environment with file '{ienv_path}' has a duplicate alias, "
+                                  "removing...")
+                else:
+                    new_servers[ienv_path] = entry
+                    if "alias" in entry:
+                        cur_aliases.add(entry["alias"])
+            self.servers = new_servers
+            if self.cur_env not in self.servers:
+                warnings.warn("Current environment is not available, switching to first available.")
+                self.cur_env = list(self.servers)[0]
+        except ValueError as exc:
+            print(exc)
+            self.reset()
+        self.save()
+
+    def save(self):
+        """Save the configuration back to the configuration file."""
+        Path(self.config_fp).parent.mkdir(exist_ok=True, parents=True)
+        with open(self.config_fp, "w", encoding="utf-8") as handle:
+            json.dump(
+                {"cur_env": self.cur_env,
+                 "servers": self.servers},
+                handle, indent=4)
+
+    def get_entry(self, path_or_alias: Union[Path, str, None] = None) -> tuple[str, dict]:
+        """Get the path and contents that belongs to a path or alias.
+
+        Parameters
+        ----------
+        path_or_alias, optional
+            Either an absolute path or an alias, by default None in which
+            case the currently selected environment is chosen.
+
+        Returns
+        -------
+        ienv_path:
+            The absolute path to the iRODS environment file.
+        entry:
+            Entry for the environment file containing, cwd, alias, cached password.
+
+        Raises
+        ------
+        KeyError
+            If the entry can't be found.
+
+        """
+        path_or_alias = self.cur_env if path_or_alias is None else path_or_alias
+        for ienv_path, entry in self.servers.items():
+            if ienv_path == str(path_or_alias):
+                return ienv_path, entry
+
+        for ienv_path, entry in self.servers.items():
+            if entry.get("alias", None) == str(path_or_alias):
+                return ienv_path, entry
+
+        raise KeyError(f"Cannot find entry with name/path '{path_or_alias}'")
```

Comment:
I think a user will never encounter this error message?! They only get the error messages then from the init, list ....

Reply:
No, they should not. It is used internally.